authors/apps/articles/serializers.py

```diff
@@ -136,23 +136,56 @@ class CommentSerializer(serializers.ModelSerializer):
     class Meta:
         model = Comment
         fields = [
-            'id', 'author', 'article', 'body',
+            'id', 'author', 'author_name', 'article', 'body',
             'parent', 'created_at', 'updated_at', 'thread_count'
         ]
 
     def create(self, validated_data):
+        author_name = self.context.get('author_name', None)
         author = self.context.get('author', None)
         article = self.context.get('article', None)
         parent = self.context.get('parent', None)
 
         return Comment.objects.create(
+            author_name=author_name,
             author=author,
             article=article,
             parent=parent,
             **validated_data
         )
 
 
+class CommentCreateSerializer(serializers.ModelSerializer):
+    author_name = serializers.SerializerMethodField()
+
+    def get_author_name(self, field):
```

Comment:
Is there a reason for the `field`  argument, @MuhanguziDavid ?

Reply:
Yes @RonKbS , the field argument takes in the author id provided as a parameter when the `get_author_name` method is called. (Look at line 175).